bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/api/dto/clip2/helper/Setters.java

```diff
@@ -168,11 +168,12 @@ public static Resource setColorTemperaturePercent(Resource target, Command comma
      * @return the target resource.
      */
     public static Resource setColorXy(Resource target, Command command, @Nullable Resource source) {
-        if (command instanceof HSBType) {
+        if (command instanceof HSBType cmdHsb) {
             Gamut gamut = target.getGamut();
             gamut = Objects.nonNull(gamut) ? gamut : Objects.nonNull(source) ? source.getGamut() : null;
             gamut = Objects.nonNull(gamut) ? gamut : ColorUtil.DEFAULT_GAMUT;
-            HSBType hsb = (HSBType) command;
+            HSBType hsb = cmdHsb.getBrightness().doubleValue() >= 0.1 ? cmdHsb
+                    : new HSBType(cmdHsb.getHue(), cmdHsb.getBrightness(), new PercentType(BigDecimal.valueOf(0.1)));
```

Comment:
Doesn't mean setting the brightness to 0.1 that the light is (slightly) on, while the desired brightness might be 0 / off?

Reply:
For my understanding, how does the value 0.1 relate to this exactly? Is it somehow deducted from the gamut?

Reply:
> how does the value 0.1 relate to this exactly?

It is an arbritrarily low number that is not exactly zero. The ColorUtil conversion algorithm divides an interim number by B, and in order to avoid a divide by zero error, the code takes a special path when B == 0 .. and this is what causes the bug. I chose 0.1 since it fulfils three criteria 1) it is not exactly 0 (thus avoid the special path), 2) it is below the minimum dimming level of all actual lamps (the lowest min dim level that I have observed is 0.2), and 3) it is not too low that the division by B in the main code path would cause a float overflow NaN result.

Reply:
.. and yes .. in case you ask .. the bug could also be fixed by changing the OH core ColorUtil code. In fact I have so far found at least five edge case bugs in ColorUtil. And so I am working on a full revamp of ColorUtil to resolve those. However since a full revamp could impact on other bindings, that work will take me several weeks if not months to complete it. Therefore I have made this particular PR to solve this one particular edge case bug in this one particular binding.